src/cursor.coffee

```diff
@@ -304,13 +304,14 @@ class Cursor extends Model
     else
       {row, column} = @getScreenPosition()
 
+      originalRow = row
       while columnCount > column and row > 0
         columnCount -= column
         column = @editor.lineTextForScreenRow(--row).length
         columnCount-- # subtract 1 for the row move
 
       column = column - columnCount
-      @setScreenPosition({row, column})
+      @setScreenPosition({row, column}, wrapAtSoftNewlines: originalRow == row)
```

Comment:
> My intuition would be that we always wrap at soft newlines when moving right, never when moving left.

That was the behavior before https://github.com/as-cii/atom/commit/e5eaa2538189ad2424fc13e6b204c10460026be5. Unfortunately, that causes an issue when moving down from a soft wrapped line which has a phantom token right under the current column (e.g. `{ currentRow + 1, currentColumn }`). Basically the cursor moves to the end of the current row, rather than to the beginning of the next row, whereas I would expect the opposite. The same happens when clicking on a phantom token: the cursor gets moved at the end of the previous row, rather than to the first *real token* of the clicked line.

An alternative could be to force `wrapAtSoftNewlines` when moving down and up, but what about the scenario in which a user clicks the aforementioned empty space (namely, the phantom token)?